### PR TITLE
Fix creating SQLServerDataTable being O(n^2) issue

### PR DESCRIPTION
**Pull request by peterbae**
Addresses issue https://github.com/Microsoft/mssql-jdbc/issues/497.

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java [hunk 1]
```diff
@@ -13,17 +13,20 @@
...
 import java.util.Map.Entry;
+import java.util.Set;
 import java.util.UUID;
 
 public final class SQLServerDataTable {
 
     int rowCount = 0;
     int columnCount = 0;
     Map<Integer, SQLServerDataColumn> columnMetadata = null;
+    Set<String> columnList = null;
```
Comment (alexnixon) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java]: I think naming this `columNames` would be clearer

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java [hunk 2]
```diff
@@ -89,11 +93,30 @@ public synchronized void addColumnMetadata(String columnName,
...
     /**
+     * Checks if duplicate columns exists, in O(n) time.
+     * 
+     * @param columnName
+     *            the name of the column
+     * @throws SQLServerException
+     *             when a duplicate column exists
+     */
+    private void checkDuplicateColumnName(String columnName) throws SQLServerException {
+        if (null != columnList) {
```
Comment (alexnixon) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java]: I'd rather an NPE than silently fail to check dupes if someone messes up null handling.

Better IMO would be defining the local variable like `Set<String> columnNames = new HashSet<>();` and omitting this check, but I don't know if that'd violate the project's coding style.

**Comment by peterbae** [comment]
@alexnixon Thanks for the review. I added the same logic for TVP since it was also applicable there, and made minor changes according to your suggestions.

**Comment by v-afrafi** [comment]
let's also add a test for checking duplicate column
